scandium-core/src/main/java/org/eclipse/californium/scandium/dtls/pskstore/BytesPskStore.java

```diff
@@ -0,0 +1,112 @@
+/*******************************************************************************
+ * Copyright (c) 2019 Bosch Software Innovations GmbH and others.
+ *
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * and Eclipse Distribution License v1.0 which accompany this distribution.
+ *
+ * The Eclipse Public License is available at
+ *    http://www.eclipse.org/legal/epl-v10.html
+ * and the Eclipse Distribution License is available at
+ *    http://www.eclipse.org/org/documents/edl-v10.html.
+ *
+ * Contributors:
+ *    Bosch Software Innovations GmbH - initial API and implementation
+ *                                      derived from PskStore
+ *******************************************************************************/
+
+package org.eclipse.californium.scandium.dtls.pskstore;
+
+import java.net.InetSocketAddress;
+
+import org.eclipse.californium.scandium.dtls.PskPublicInformation;
+import org.eclipse.californium.scandium.util.ServerNames;
+
+/**
+ * A service for resolving pre-shared key identities.
+ */
+public interface BytesPskStore {
+
+	/**
+	 * Gets the shared key for a given identity.
+	 * <p>
+	 * A DTLS server can use this method to look up the pre-shared key for an
+	 * identity provided by the client as part of a PSK key exchange.
+	 * </p>
+	 * <p>
+	 * The implementation is intended to normalize the identity by a matching
+	 * entry, if that entry is not UTF-8 compliant encoded.
+	 * </p>
```

Comment:
I spend some times to understand the idea behind normalizing. This is mainly for clearer log, right ?
Maybe we could explain more about that and maybe adding an example of value for pskPublicinformation for not UTF-8 compliant encoded in javadoc could help to better understand the idea.

Reply:
No, it's mainly for the principal `PreSharedKeyIdentity`, that's the major downside of "non-compliant" clients. The principal requires a proper, unique name.

Reply:
OK, so this confirms this deserve more explanation :sweat_smile:

Reply:
I try it.